templates/repo/settings/branches.tmpl

```diff
@@ -77,6 +77,28 @@
 					</div>
 				</div>
 			</div>
+
+			{{if $.Repository.CanCreateBranch}}
+				<h4 class="ui top attached header">
+					{{.i18n.Tr "repo.settings.rename_branch"}}
+				</h4>
+				<div class="ui attached segment">
+					<form class="ui form" action="{{$.Repository.Link}}/settings/rename_branch" method="post">
```

Comment:
That is one approach.
I would have guessed that you choose a branch, then a dialog pops up where the base branch is already set, and then you enter the target branch name.
But this approach should also work, but might be a bit less user-friendly.